### ISSUE TITLE
Adding stretch to a widget's native layout causes a crash

### ISSUE THREAD

**Issue by jni** [napari/magicgui#111]
**Describe the bug**

When I try to add Qt (native) stretch to a magicgui widget using `func.native_layout.addStretch()`, I get a crash with the following traceback:

```pytb
 $ python -m affinder.affinder
WARNING: Traceback (most recent call last):
  File "/Users/jni/conda/envs/all/lib/python3.8/site-packages/magicgui/widgets/_bases.py", line 245, in _emit_parent
    self.parent_changed(value=self.parent)
  File "/Users/jni/conda/envs/all/lib/python3.8/site-packages/magicgui/events.py", line 603, in __call__
    self._invoke_callback(cb, event)
  File "/Users/jni/conda/envs/all/lib/python3.8/site-packages/magicgui/events.py", line 621, in _invoke_callback
    _handle_exception(
  File "/Users/jni/conda/envs/all/lib/python3.8/site-packages/magicgui/events.py", line 619, in _invoke_callback
    cb(event)
  File "/Users/jni/conda/envs/all/lib/python3.8/site-packages/magicgui/widgets/_bases.py", line 1102, in reset_choices
    for widget in self:
  File "/Users/jni/conda/envs/all/lib/python3.8/_collections_abc.py", line 874, in __iter__
    v = self[i]
  File "/Users/jni/conda/envs/all/lib/python3.8/site-packages/magicgui/widgets/_bases.py", line 996, in __getitem__
    item = self._widget._mgui_get_index(key)
  File "/Users/jni/conda/envs/all/lib/python3.8/site-packages/magicgui/backends/_qtpy/widgets.py", line 276, in _mgui_get_index
    return item.widget()._magic_widget
AttributeError: 'NoneType' object has no attribute '_magic_widget'
Abort trap: 6
/Users/jni/conda/envs/all/lib/python3.8/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
```

**To Reproduce**
```
pip install napari[all] git+https://github.com/jni/affinder@layout-stretch
python -m affinder.affinder
```

The commit causing issues is here:

https://github.com/jni/affinder/commit/b45a97cee35a48d8eb9da8049151ffd945cb2f5d

**Expected behavior**
stretch would be added to the bottom of the widget to make the layout prettier. 😊 

**Screenshots**

**Environment (please complete the following information):**
 - OS: [e.g. Windows, Mac] Mac
 - backend: [e.g. Qt 5.13.0] PyQt5: 5.14.2
 - magicgui version [e.g. 0.1.0] 0.2.5

**Comment by tlambert03** [comment]
ok, so it's easy enough to fix this particular bug in one line, but it exposes an issue if someone starts adding native Qt widgets directly to `widget.native_layout` (putting this here for my own sake as much as yours).  `Container` has been [setup up as a mutable sequence](https://github.com/napari/magicgui/blob/34bad9687a46747b37b0d3d65e2b95c5130c712e/magicgui/widgets/_bases.py#L903) of `magicgui.widgets.Widget`.  This makes for that nice API of adding/removing from layouts with basic python syntax.  On the backend, we add the _native_ widget to the layout, but as we retrieve them when indexing into the container, we grab the `_magic_widget` instance that has been stored on all native widgets.

However, the length of that "list" is [based on the number of items native layout](https://github.com/napari/magicgui/blob/34bad9687a46747b37b0d3d65e2b95c5130c712e/magicgui/widgets/_bases.py#L1054).  So if someone starts adding to native_layout directly, then we _either_ have to deal with the fact that `Container` is now a mutable sequence of `Union[magicgui.Widget, backend.widget.class]`, or deal with index errors.  neither of those are great.

This is a tough one, because I don't want straight up break the possibility of using the native API in certain cases (for those who know how to use it), but I also want to keep a strict separation between backend and magicgui APIs.  Maybe the best option for now is to decouple the length of the `Container` from the length of the native layout, and effectively just "hide" things that have been added to the native layout directly.

@jni?

**Comment by tlambert03** [comment]
note to @jni: the fix in #113 would require you to change your affinder line to `start_affinder.native.layout().addStretch()`
